tests/unit/optional-tag-creation.test.mjs

```diff
@@ -0,0 +1,152 @@
+import { describe, expect, test, jest, beforeEach } from '@jest/globals';
+import { createGitWorkflowMock, createInquirerWorkflowMock } from '../mocks/workflow-mock.mjs';
+
+// Mock branches-actions module
+jest.unstable_mockModule('../../commands/branches-actions.mjs', () => {
```

Comment:
This test does not do anything as it mocks the entire workflow.

Reply:
We should mock the functions inside but use action doRelease to see if the `createTag` is actually run or not